src/realm/array.cpp

```diff
@@ -1717,9 +1717,9 @@ void Array::alloc(size_t init_size, size_t width)
             // field in the header.
             size_t new_capacity_bytes = capacity_bytes * 2;
             if (new_capacity_bytes < capacity_bytes) // overflow detected, clamp to max
```

Comment:
Should this check actually be removed? `new_capacity_bytes` can only be less than `capacity_bytes` if the multiplication overflows, and that can only happen if `capacity_bytes` is bigger than half of what can be stored in `size_t`, but `capacity_bytes` uses at most 24 bits.